app/models/concerns/earliest_date_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Validates that a given date (attribute) does not occur before the date provided in options
+class EarliestDateValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return unless options.key?(:date)
+
+    raise TypeError, "#{options[:date]} does not implement required comparison methods" unless options[:date].acts_like?(:date)
+
+    record.errors.add(attribute, "cannot occur before #{options[:date]}") unless valid?(value, options[:date])
+  end
+
+  private
+
+  def  valid?(value, provided)
```

Comment:
minor thing, but is there a reason for the double space here or was it a typo? same thing in `latest_date_validator`